arbnode/espresso_caff_node.go

```diff
@@ -160,13 +162,29 @@ func NewEspressoCaffNode(
 		log.Crit("Failed to create hotshot client", "err", err)
 	}
 
+	fromBlock := configFetcher().FromBlock
+	if !configFetcher().Dangerous.IgnoreDatabaseFromBlock {
+		fromBlock, err = readCurrentFromBlockFromDb(db)
+		if err != nil {
+			log.Crit("failed to read l1 block from db", "err", err)
+		}
+	}
+
+	if fromBlock == 0 {
+		fromBlock = configFetcher().FromBlock
+		if fromBlock == 0 {
+			log.Crit("fromBlock is 0, please provide a valid block number")
+		}
+	}
```

Comment:
Idk if this is cleaner or not, but couldn't we do something like:

```suggestion
	fromBlock uint64
	if !configFetcher().Dangerous.IgnoreDatabaseFromBlock {
		fromBlock, err = readCurrentFromBlockFromDb(db)
		if err != nil {
			log.Crit("failed to read l1 block from db", "err", err)
		}
	}
    
	if fromBlock == 0 && configFetcher().FromBlock != 0{
		fromBlock = configFetcher().FromBlock
	} else {
	    log.Crit("configured fromBlock is 0, please provide a valid block number")
	}
```